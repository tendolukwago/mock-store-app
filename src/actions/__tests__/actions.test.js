
//third party imports
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
//local imports
import * as actions from '../index';

const baseUrl = 'http://localhost:8080';

const middlewares = [thunk]; 
const mockStore = configureStore(middlewares);

describe('Actions for Fetch', () => {

    const initialState = {};       
    const token = "token";  
    const store = mockStore(initialState);

    beforeEach(() => {

        store.clearActions();
    });


     //Fetch Posts
    describe('Fetch', () => {
        it('creates a FETCH action', async () => {

            //arrage

            //act
            nock(baseUrl)
            .get('/api/test')
            .reply(200);

            await store.dispatch(actions.fetch(token));

            //assert
            expect(store.getActions()).toMatchSnapshot();

        });
        it('creates a FETCH_SUCCESS action', () => {
            //arrange
 
            const res = {data: [
                {
                    id:"1",
                    title:"title",
                    body:"body"
                },
                {
                    id:"2",
                    title:"title",
                    body:"body"
                }
            ]};
            //act
            store.dispatch(actions.fetchSuccess(res));

            //assert
            expect(store.getActions()).toMatchSnapshot();
        });
        it('creates a FETCH_FAILURE action', () => {
            //arrage
            const store = mockStore(initialState);
            const err = {message:"internal server error"};

            //act
            store.dispatch(actions.fetchFailure(err));

            //assert
            expect(store.getActions()).toMatchSnapshot();
        });
        it('creates a RESET_FETCH action', () => {
            //arrage


            //act
            store.dispatch(actions.resetFetch());

            //assert
            expect(store.getActions()).toMatchSnapshot();   
        });
    });
});