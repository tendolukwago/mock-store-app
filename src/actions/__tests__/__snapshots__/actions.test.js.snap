// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Actions for Fetch Fetch creates a FETCH action 1`] = `
Array [
  Object {
    "payload": Promise {
      "_40": 0,
      "_55": Promise {
        "_40": 0,
        "_55": null,
        "_65": 0,
        "_72": null,
      },
      "_65": 3,
      "_72": null,
    },
    "type": "FETCH",
  },
]
`;

exports[`Actions for Fetch Fetch creates a FETCH_FAILURE action 1`] = `
Array [
  Object {
    "payload": Object {
      "message": "internal server error",
    },
    "type": "FETCH_FAILURE",
  },
]
`;

exports[`Actions for Fetch Fetch creates a FETCH_SUCCESS action 1`] = `
Array [
  Object {
    "payload": Object {
      "data": Array [
        Object {
          "body": "body",
          "id": "1",
          "title": "title",
        },
        Object {
          "body": "body",
          "id": "2",
          "title": "title",
        },
      ],
    },
    "type": "FETCH_SUCCESS",
  },
]
`;

exports[`Actions for Fetch Fetch creates a RESET_FETCH action 1`] = `
Array [
  Object {
    "type": "RESET_FETCH",
  },
]
`;
